core/genesis.go

```diff
@@ -328,7 +328,7 @@ func (g *Genesis) toBlock(db ethdb.Database, triedb *triedb.Database) *types.Blo
 		}
 	}
 
-	statedb.Commit(0, false, false)
+	statedb.Commit(0, false)
```

Comment:
good point, but it ends up not being an issue:
- in most cases in the next line triedb.Commit will be called, so the root will be removed from dirties (the ref-counted map) and put on disk
- if the genesis is empty I think it's fine because there would be no call to Update (because of root != origin check in statedb)